app/routes/transfer/gift/$chainId.$assetId/amount.tsx

```diff
@@ -1,42 +1,68 @@
+import { useNavigate } from 'react-router-dom';
+
 import { Button, Progress } from '@nextui-org/react';
-import { $path } from 'remix-routes';
+import { type ClientLoaderFunction, useLoaderData } from '@remix-run/react';
+import { $params, $path } from 'remix-routes';
 
-import { useMainButton } from '@/common/telegram/useMainButton';
+import { useAmountLogic } from '@/common/_temp_hooks/useAmountLogic';
+import { useGlobalContext } from '@/common/providers';
+import { BackButton } from '@/common/telegram/BackButton';
+import { MainButton } from '@/common/telegram/MainButton';
+import { pickAsset } from '@/common/utils';
 import { BodyText, HeadlineText, Icon } from '@/components';
+import { AmountDetails } from '@/components/AmountDetails';
+
+export const clientLoader = (({ params }) => {
+  return $params('/transfer/gift/:chainId/:assetId/amount', params);
+}) satisfies ClientLoaderFunction;
+
+const Page = () => {
+  const { chainId, assetId } = useLoaderData<typeof clientLoader>();
 
-import AmountDetails from './AmountDetails';
-import { useAmountLogic } from './useAmountLogic';
+  const { assets } = useGlobalContext();
+  const navigate = useNavigate();
+
+  const selectedAsset = pickAsset(chainId, assetId, assets);
 
-export default function AmountGiftPage() {
-  const { mainButton } = useMainButton();
   const {
     handleMaxSend,
     handleChange,
     setIsAmountValid,
-    isAccountTerminate,
+    getIsAccountToBeReaped,
     isPending,
     deposit,
-    selectedAsset,
     amount,
+    fee,
     maxAmountToSend,
     isAmountValid,
   } = useAmountLogic({
-    prevPage: $path('/transfer/select-token'),
-    nextPage: $path('/transfer/gift/create'),
-    mainButtonText: 'Enter Amount',
-    onAmountChange: () => {
-      mainButton.setText('Create gift');
-      setIsAmountValid(prev => prev && !!deposit && +amount >= deposit);
-    },
+    selectedAsset,
+    // onAmountChange: () => {
+    //   setIsAmountValid(prev => prev && !!deposit && +amount >= deposit);
+    // },
```

Comment:
this part was needed to check for gifts that a transfer amount was bigger than a deposit. Otherwise, a gift account cannot be created. 
You can move this condition to another place since we no longer have a dependence on mainbatton